### PR TITLE
Adding Text Encoder/Decoder fallback tests

### PR DESCRIPTION
**Pull request by tarekgh**

### CODE REVIEW

File: src/System.Text.Encoding.Extensions/tests/System.Text.Encoding.Extensions.Tests.csproj [hunk 1]
```diff
@@ -25,6 +25,7 @@
     <Compile Include="UTF32LE.cs" />
     <Compile Include="UTF7.cs" />
     <Compile Include="UTF8.cs" />
+    <Compile Include="fallback.cs" />
```
Comment (stephentoub) [inline src/System.Text.Encoding.Extensions/tests/System.Text.Encoding.Extensions.Tests.csproj]: Nit: Capital 'F' both here and for the actual file name?

File: src/System.Text.Encoding.Extensions/tests/fallback.cs [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+        [Fact]
+        public static void TestEncoderReplacementFallback()
+        {
+            Encoding asciiEncoding = Encoding.GetEncoding("us-ascii", new EncoderReplacementFallback("(unknown)"), new DecoderReplacementFallback(""));
+            byte[] encodedBytes = new byte[asciiEncoding.GetByteCount(s_asciiInputStringWithFallback)];
+
+            int numberOfEncodedBytes = asciiEncoding.GetBytes(s_asciiInputStringWithFallback, 0, s_asciiInputStringWithFallback.Length, encodedBytes, 0);
+
+            AssertEqualList(
```
Comment (stephentoub) [inline src/System.Text.Encoding.Extensions/tests/fallback.cs]: Why not just Assert.Equal?

Comment (tarekgh) [inline src/System.Text.Encoding.Extensions/tests/fallback.cs]: I thought Assert.Equal is not testing the collection contents. I searched the code and I saw some other places doing the same as I did. I'll try your idea.

File: src/System.Text.Encoding.Extensions/tests/System.Text.Encoding.Extensions.Tests.csproj (hunk 1 above)
Comment (tarekgh) [inline src/System.Text.Encoding.Extensions/tests/System.Text.Encoding.Extensions.Tests.csproj]: will do that

**Comment by tarekgh** [comment]
@dotnet-bot could you please check this?

**Comment by stephentoub** [comment]
@dotnet-bot test this please

File: src/System.Text.Encoding.Extensions/tests/Fallback.cs [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+using Xunit;
+using System;
+using System.Text;
+
+namespace EncodingTests
+{
+    public static class Fallback
+    {
+        static readonly string s_asciiInputStringWithFallback = "\u00abX\u00bb"; // DOUBLE ANGLE QUOTATION MARK (U+00AB), 'X' (U+0058), and RIGHT POINTING DOUBLE ANGLE QUOTATION MARK (U+00BB).
+        static readonly string s_asciiInputStringWinNoFallback = "abc";
```
Comment (justinvp) [inline src/System.Text.Encoding.Extensions/tests/Fallback.cs]: nit: specify `private` visibility

File: src/System.Text.Encoding.Extensions/tests/Fallback.cs [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+
+    // a simple fallback class which fallback using the character sequence 'a', 'b', 'c',...etc.
+    internal sealed class EncoderCustomFallback : EncoderFallback
+    {
+        public EncoderCustomFallback()
+        {
+        }
+
+        public String DefaultString
```
Comment (justinvp) [inline src/System.Text.Encoding.Extensions/tests/Fallback.cs]: nit: use `string` keyword vs type name.

File: src/System.Text.Encoding.Extensions/tests/Fallback.cs [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        public override int GetHashCode()
+        {
+            return "EncoderCustomFallback".GetHashCode();
+        }
+    }
+
+    internal sealed class EncoderCustomFallbackBuffer : EncoderFallbackBuffer
+    {
+        int nextChar;
+        bool fallback;
```
Comment (justinvp) [inline src/System.Text.Encoding.Extensions/tests/Fallback.cs]: nit: specify `private` visibility

Comment (justinvp) [inline src/System.Text.Encoding.Extensions/tests/Fallback.cs]: nit: also, field names should start with underscore

**Pull request merged**